src/js/view/controls/controls.js

```diff
@@ -1,31 +1,386 @@
         }
 
-        enable() {
+        enable(api, model, videoLayer) {
+            // Dock Area and Buttons
+            if (!this.dock) {
+                this.dock = new Dock(model);
+            }
+            this.right.appendChild(this.dock.element());
+
+            // Display Buttons
+            if (!this.displayContainer) {
+                const displayContainer = new DisplayContainer();
+                const rewindDisplayIcon = new RewindDisplayIcon(model, api);
+                const playDisplayIcon = new PlayDisplayIcon(model);
+                // toggle playback
+                playDisplayIcon.on('click tap', () => {
+                    this.trigger(events.JWPLAYER_DISPLAY_CLICK);
+                    this.userActive(1000);
+                    api.play(reasonInteraction());
+                });
+                // make playDisplayIcon clickthrough on chrome for flash to avoid power safe throttle
+                if (utils.isChrome() && !utils.isMobile()) {
+                    playDisplayIcon.el.addEventListener('mousedown', () => {
+                        const provider = model.getVideo();
+                        const isFlash = (provider && provider.getName().name.indexOf('flash') === 0);
+                        if (!isFlash) {
+                            return;
+                        }
+                        const resetPointerEvents = function() {
+                            document.removeEventListener('mouseup', resetPointerEvents);
+                            playDisplayIcon.el.style.pointerEvents = 'auto';
+                        };
+                        this.style.pointerEvents = 'none';
+                        document.addEventListener('mouseup', resetPointerEvents);
+                    });
+                }
+                const nextDisplayIcon = new NextDisplayIcon(model, api);
+                displayContainer.addButton(rewindDisplayIcon);
+                displayContainer.addButton(playDisplayIcon);
+                displayContainer.addButton(nextDisplayIcon);
+                this.element.appendChild(displayContainer.element());
+                this.displayContainer = displayContainer;
+            }
+
+
+            // Display Click and Double Click Handling
+            const displayClickHandler = new ClickHandler(model, videoLayer, { useHover: true });
+            displayClickHandler.on({
+                click: () => {
+                    this.trigger(events.JWPLAYER_DISPLAY_CLICK);
+                    api.play(reasonInteraction());
+                },
+                tap: () => {
+                    this.trigger(events.JWPLAYER_DISPLAY_CLICK);
+                    const state = model.get('state');
+
+                    if (((state === states.IDLE || state === states.COMPLETE) ||
+                        (this.instreamState === states.PAUSED))) {
+                        api.play(reasonInteraction());
+                    }
+                    if (state === states.PAUSED) {
```

Comment:
If we're `PAUSED` and  instreamState is `PAUSED` are we supposed to enter both `if`s?